### DISCUSSION TITLE
Set multiple datacontext for template control?

### DISCUSSION THREAD

**Discussion by XDgl233** [AvaloniaUI/Avalonia#12265] [Q&A] [answered]
Hi! I am currently working on a template control, it's a composite button with buttons, labels, and borders. So far I have the animations and custom properties binding working without problems.
But I encounter this problem: one of the labels will show the total uptime of the program, and the other label will show whether the port is open and flip the chained ellipse's color. I get these two pieces of information (uptime and whether the port is open or not) from different classes, and I can get one of them working if I set the data context of the whole template control to that corresponding class (ObservableObject as a partial class, using Community Toolkit MVVM).

So naturally, I think setting data context to these different labels can solve the problem, but no success. I tried using <customcontrol_xxx.style> and style setter to set both the data context and content properties, but it seems the dependency is wrong and the content binding can't find the target property from the data context set in this way. What should I do in these situations, do I need to make a new class to gather all the needed properties?

Another question, during working with templates, I am also wondering if it's possible that I can use the template control while directly access to the child control? I want to add some control for example a label directly into the child control say, a grid in that template control, under certain conditions. I just use a workaround that essentially duplicate the whole template style and change different control and structures, but that's time-consuming and somewhat against the idea of using template in the first place.

Thank you.

**Comment by timunie** [comment]
See:

- https://docs.avaloniaui.net/docs/next/basics/data/data-binding/data-context
- https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/CustomControls/RatingControlSample

**Reply by XDgl233** [reply, marked as answer]
Thank you, I changed my idea and use content presenters with styled elements to achieve different loadouts.
